src/usdb_syncer/song_loader.py

```diff
@@ -668,6 +711,57 @@ def _maybe_write_audio_tags(ctx: _Context) -> None:
     )
 
 
+def _maybe_separate_stems(ctx: _Context) -> None:
+    if not (audio_options := ctx.options.audio_options) or (
+        audio_options.stem_separation == AudioStemSeparation.DISABLE
+    ):
+        return
+    if not (ctx.song.sync_meta):
+        return
+
+    model = audio_options.stem_separation.value
+    demucs.separate.main([
+        "--two-stems",
+        "vocals",
+        "-n",
+        model,
+        f"{ctx.out.audio.path(ctx.locations, temp=True)}",
+        "-o",
+        f"{ctx.locations._tempdir}",
+    ])
+    # Transcode instrumental and vocals files to target format via ffmpeg
+    instrumental_input = str(ctx.locations.stem_separation_instrumental_path(model))
+    instrumental_output = (
+        f"{ctx.locations.temp_path()} [INSTR].{audio_options.format.value}"
+    )
+    transcode_audio(audio_options, instrumental_input, instrumental_output)
+    ctx.out.instrumental.resource = ctx.out.audio.resource
+    ctx.out.instrumental.new_fname = (
+        f"{ctx.locations.filename()} [INSTR].{audio_options.format.value}"
+    )
+
+    vocals_input = str(ctx.locations.stem_separation_vocals_path(model))
+    vocals_output = f"{ctx.locations.temp_path()} [VOC].{audio_options.format.value}"
+    transcode_audio(audio_options, vocals_input, vocals_output)
+    ctx.out.vocals.resource = ctx.out.audio.resource
+    ctx.out.vocals.new_fname = (
+        f"{ctx.locations.filename()} [VOC].{audio_options.format.value}"
```

Comment:
Isn't this just `vocals_output.name`?

Reply:
You mean `Path(vocals_output).name`, right?

Reply:
Ah, yes. But I would build the correct path object first: `vocals_output = ctx.locations.temp_path(ext=f" [VOC].{audio_options.format.value}")`.